addons/web/views/database_manager.html

```diff
@@ -53,27 +53,69 @@
                         }
                     }
                 });
+
+                // generate a random master password
+                // removed lI1oO0 to avoid confusions
+                var charset = "abcdefghijkmnpqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ23456789";
+                var password = "";
+                for (var i = 0, n = charset.length; i < 12; ++i) {
+                    password += charset.charAt(Math.floor(Math.random() * n));
+                    if (i === 3 || i === 7) {
+                        password += "-";
+                    }
+                }
+                var master_pwds = document.getElementsByClassName("generated_master_pwd");
+                for (var i=0, len=master_pwds.length|0; i<len; i=i+1|0) {
+                    master_pwds[i].innerText = password;
+                }
+                var master_pwd_inputs = document.getElementsByClassName("generated_master_pwd_input");
+                for (var i=0, len=master_pwd_inputs.length|0; i<len; i=i+1|0) {
+                    master_pwd_inputs[i].value = password;
+                    master_pwd_inputs[i].setAttribute('autocomplete', 'new-password');
+                }
             });
         </script>
     </head>
 
-    {% macro master_input() -%}
-        <div class="form-group">
-            {% if insecure %}
+    {% macro master_input(allow_insecure=False) -%}
+
+        {% set input_class = "form-control" %}
+        {% if insecure %}
+            {%if allow_insecure %}
                 <input type="hidden" name="master_pwd" class="form-control" value="admin"/>
             {% else %}
-                <label for="master_pwd" class="col-form-label">Master Password</label>
-                <input id="master_pwd" type="password" name="master_pwd" class="form-control" required="required" autofocus="autofocus"/>
+                <div class="alert alert-warning">
+                    <p>Warning, your Odoo database manager is not protected. To secure it, we have generated the following master password for it:</p>
+                    <p style="text-align: center;"><strong class="generated_master_pwd"></strong></p>
+                    <p>You can change it below but be sure to remember it, it will be asked for future operations on databases.</p>
+                </div>
+                {% set input_class = "form-control generated_master_pwd_input" %}
             {% endif %}
-        </div>
+        {% endif %}
+        {% if not insecure or not allow_insecure %}
```

Comment:
how can you change your insecure password if you don't show the input?

Reply:
If it's insecure and you used the "Set Master Password" button, it's the `type="hidden"` above (and you set the new master password in the `master_pwd_new` input below). If you click on another button, `allow_insecure` is `False` so you see the input. I will rename the variable to make it clearer.